core/server/models/plugins/pagination.js

```diff
@@ -130,6 +128,11 @@ pagination = function pagination(bookshelf) {
         /**
          * ### Fetch page
          * A `fetch` extension to get a paginated set of items from a collection
+         *
+         * We trigger two queries:
+         * 1. count query to know how many pages left (important: we don't attach any group/order statements!)
+         * 2. the actualy fetch query with limit and page property
```

Comment:
the only change in this file, which was fundamental, is: to separate the count and the fetch query for pagination. the reason is that postgres and mysql/sqlite behave differently when using group and order statements. in the case of counting the posts, we don't need **any** order/group statements, which helped fixing the tests for postgres, because postgres was throwing errors that we are misusing the queries.

see also https://github.com/tgriesser/bookshelf/blob/master/src/plugins/pagination.js#L168

There is a complete `fetchPage` method in `bookshelf`, i don't know why we are not using it?if somebody can tell me why, i could add a note here.
see http://bookshelfjs.org/#Model-instance-fetchPage